### ISSUE TITLE
Undefined function or variable 'dd'

### ISSUE THREAD

**Issue by brosscle** [WuChanada/StripSkullCT#2]
Hi, I am using your code to perform a brain extraction on CT images but I sometimes get stuck with an issue in the SkullStrip function, which is private so I cannot investigate. The error message is : 
`Error in SkullStrip (line 10)
Unrecognized function or variable 'dd'`

Do you have any hint about how to solve this issue ?
Thanks a lot !
Clément

**Comment by Uni-Summer** [comment]
hey,
sorry to bother,
but have you solved this error? I just met the same problem.
It would help a lot if you could reply. Thanks so much.
Best.
Ziying

**Comment by brosscle** [comment]
Hi,
If I remember well, it was because the resulting extracted brain was empty, due to an error in the input image (for instance a badly converted nifti image containing a deformed brain or no brain at all).
However, since the function is private I couldn't resolve it properly.
Hope it will help you !
Best, 
Clément

**Comment by Uni-Summer** [comment]
Thanks so much for your reply!
If I may ask, have you tried any other skull-stripping algorithms to perform well on CT? As a  newbie, your advice can really help me.
Best.
Ziying

**Comment by brosscle** [comment]
I tried the "classic" ones for MRI: FSL BET and PCNN3D, but I ended up using this script because I found the result a bit more robust. However those tests took place 3/4 years ago so other algorithms might have emerged, especially deep learning based ones.
If I had to do it again, and if no new deep learning algorithm suits me, I would probably try building my own brain extraction tool by taking advantage of the quantitative values of CT. I would probably begin by a thresholding the images between -15 and 150 HU and then apply image processing operations as morphological or "keep largest components" operations.
Hope this helps !
Best,
Clément

**Comment by Uni-Summer** [comment]
yes, I also tried FSL BET, and it performs badly in my case. however, this code only suits part of my dataset, it went wrong with some data. And the value of HU has been clipped to [0, 100] before input. 
Still, thank you so much for your advice, I will try more methods. 
Best,
Ziying
